cli/azd/cmd/up.go

```diff
@@ -99,8 +109,17 @@ func (u *upAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 	}
 	defer func() { _ = infra.Cleanup() }()
 
+	// TODO(weilim): remove this once we have decided if it's okay to not set AZURE_SUBSCRIPTION_ID and AZURE_LOCATION
+	// early in the up workflow in #3745
 	err = u.provisioningManager.Initialize(ctx, u.projectConfig.Path, infra.Options)
-	if err != nil {
+	if errors.Is(err, bicep.ErrEnsureEnvPreReqBicepCompileFailed) {
+		// If bicep is not available, we continue to prompt for subscription and location unfiltered
+		err = provisioning.EnsureSubscriptionAndLocation(ctx, u.envManager, u.env, u.prompters,
+			func(_ account.Location) bool { return true })
+		if err != nil {
+			return nil, err
+		}
+	} else if err != nil {
```

Comment:
I'm having a hard time understanding what the issue is.  First off - Since `up` is now a workflow and can run any azd commands, should the provisioning prompting only in `azd provision` as normal and let `up` just do its thing?

Next - Agreed that subscription and location are super important for bicep - so why aren't those captured first in bicep initialize before we attempt to compile the bicep template?

After those are collected - if there is a bicep compile issue then it feels like it should be a hard stop, report the compile failure back to the user so they can fix it??

Am I missing anything?

Reply:
>  Since up is now a workflow and can run any azd commands, should the provisioning prompting only in azd provision as normal and let up just do its thing?

I agree with this, and this is where I would like to go as well, but @vhvb1989 was slightly worried about the behavioral change because `AZURE_SUBSCRIPTION_ID` and `AZURE_LOCATION` won't be available at the start of the `up` workflow, which is observable by hooks (example a `preprovision` hook).

> After those are collected - if there is a bicep compile issue then it feels like it should be a hard stop, report the compile failure back to the user so they can fix it??

A bicep compile should be a hard stop. The soft-continue added here is just to continue the existing behavior in `up`, which is done to avoid `up` failing when a user doesn't have any `infra` defined.